presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CountConstantOptimizer.java

```diff
@@ -39,6 +39,7 @@
 import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
 import static java.util.Objects.requireNonNull;
 
+@Deprecated
```

Comment:
I thought more about iterative optimizer recently and I came to conclusion that current approach with feature toggle does not have much sense. Notice when iterative optimizer is enabled and it has a bug (like it does nothing) then legacy rule will eventually do a rewrite anyway. 

From the point of view of user it not possible to tell if new optimizer did given rewrite or legacy. Though, changing the toggle should force to prefer new optimizer rewrites.

With above said, how would you tell that given rule is properly migrated. To me current tests results are not enough as they do not tell which rule did the rewrite. Without such certainty we cannot eventually remove legacy (deprecated) rules.

I see two options here: 
 - remove the toggle. When rule is migrated then it should be simply removed. This makes testing and whole migration process much easier (as we test the same thing what is used on production (legacy)).  Though, it would require that rules are migrated slowly, to see interactions between them on production.
- when a rule is migrated, then legacy rule should be disabled when toggle is enabled.